### ISSUE TITLE
Caught 502 bad gateway when running by kubernetes

### ISSUE THREAD

**Issue by nautilux2** [vinbigdata-medical/vindr-lab-deployment#3]
![image](https://user-images.githubusercontent.com/6006914/113682313-35440d80-96ed-11eb-9177-03f75b2d1f37.png)
![image](https://user-images.githubusercontent.com/6006914/113682375-45f48380-96ed-11eb-9d4a-6c75e881a8bc.png)
![image](https://user-images.githubusercontent.com/6006914/113682502-6fadaa80-96ed-11eb-9623-ceee5f4eeb88.png)
![image](https://user-images.githubusercontent.com/6006914/113682593-848a3e00-96ed-11eb-9788-fd4c1156b23d.png)

Same as error occur when running by docker
I think it's some kind of misconfigured

**Comment by iamatsundere** [comment]
I've updated the Note/README in the commit [a056252](https://github.com/vinbigdata-medical/vindr-lab-deployment/commit/a0562526658652cff7bcbac4f80d4789ca077e76). Please check it.

**Comment by nautilux2** [comment]
@iamatsundere nah, I've just deployed and waited about 10mins but it still the same for keycloak
![image](https://user-images.githubusercontent.com/6006914/113805873-830f5300-978b-11eb-9058-d53dbba448e0.png)

**Comment by iamatsundere** [comment]
Have you tried something like `localhost:8080/minio` or did you use 8080 port for other service in your machine?

**Comment by nautilux2** [comment]
Hi @iamatsundere 
![image](https://user-images.githubusercontent.com/6006914/114134508-11253e00-9932-11eb-8a96-d03aaee95098.png)
Here is `localhost:8080/minio`
![vin git 2](https://user-images.githubusercontent.com/6006914/114134847-97da1b00-9932-11eb-918c-74a25735ed57.png)
And here is 8080 port

**Comment by iamatsundere** [comment]
Can you run `kubectl get svc -n vinlab` and send me the return?

**Comment by nautilux2** [comment]
![image](https://user-images.githubusercontent.com/6006914/114139734-9b24d500-9939-11eb-8e22-e7049ee09a6a.png)
Here you go

**Comment by iamatsundere** [comment]
Please jump into the `nginx` pod, then run `curl -i http://es.vinlab:9200` and check the result. I guess there might be something wrong when connect between the gateway and the services behind.

**Comment by iamatsundere** [comment]
> ![image](https://user-images.githubusercontent.com/6006914/114139858-c14a7500-9939-11eb-8b8c-e21a2a7246ea.png)
> 
> Here you go

It seems like the services in cluster worked. Please jump into the nginx pod, then run curl -i http://es.vinlab:9200 and check the result. I guess there might be something wrong when connect between the gateway and the services behind.

**Comment by nautilux2** [comment]
In the first comment, I gave you an image with an error that can not connect to elasticsearch
![image](https://user-images.githubusercontent.com/6006914/114141853-5c444e80-993c-11eb-8aed-3f5fbe7f112c.png)
I think you missed it

**Comment by iamatsundere** [comment]
> In the first comment, I gave you an image with an error that can not connect to elasticsearch
> ![image](https://user-images.githubusercontent.com/6006914/114141853-5c444e80-993c-11eb-8aed-3f5fbe7f112c.png)
> I think you missed it

My bad, I think the kubernetes `coredns` service can't resolve the services now. Can you check the log of it and restart it?

**Comment by nautilux2** [comment]
Here is coredns log
![image](https://user-images.githubusercontent.com/6006914/114143349-4041ac80-993e-11eb-87ce-83a762dd6d95.png)
Let me restart and let you know

**Comment by nautilux2** [comment]
@iamatsundere 
Dashboard url seems working coz it redirects to `http://localhost:8080/auth/realms/vinlab/protocol/openid-connect/auth?client_id=vinlab-frontend&response_type=code&state=lwvqhedf9fk&scope=profile&redirect_uri=http://localhost:8080/dashboard` but the keycloak URL still stuck in bad gateway

**Comment by iamatsundere** [comment]
I think the Keycloak works fine, because when the dashboard redirects to `http://localhost:8080/auth/realms/vinlab/protocol/openid-connect/auth?client_id=vinlab-frontend&response_type=code&state=lwvqhedf9fk&scope=profile&redirect_uri=http://localhost:8080/dashboard`, it is the link provided by Keycloak. Please retry in fresh browser and I'll think about it.

Ping me if MinIO and OrthanC url work...

**Comment by nautilux2** [comment]
@iamatsundere Neither MinIO nor OrthanC url work. Both produce 502 error code

**Comment by iamatsundere** [comment]
Can you check the nginx log or your local machine firewall?

**Comment by nautilux2** [comment]
Here nginx logs for MinIO and OrthanC
![image](https://user-images.githubusercontent.com/6006914/114489917-2655e180-9c3e-11eb-9c2c-f2fa146d3f5e.png)

**Comment by iamatsundere** [comment]
I've updated the documentations, please try it again, thanks!

**Comment by Stephen-Lobo** [comment]
Hi, I have tried to install multiple times through Docker, but getting same 502 Bad gateway error for myipaddress:8080/auth url. As per your document, I even waited for hours, but still same error.

Please help.

**Comment by iamatsundere** [comment]
Sorry for disconnecting, have you tried `http://your_host:8080/minio` or `http://your_host:8080/elasticsearch`?
